pkgs/top-level/all-packages.nix

```diff
@@ -11364,10 +11364,11 @@ in
 
   ccacheStdenv = lowPrio (makeOverridable ({ extraConfig, stdenv }:
     overrideCC stdenv (buildPackages.ccacheWrapper.override {
-      inherit extraConfig;
       inherit (stdenv) cc;
+    } // lib.optionalAttrs (extraConfig != null) {
```

Comment:
In my opinion, `null' is not equivalent to an empty string. I think there may be some problems here.
How about comparing it to an empty string? Or is this not required and everything will work?

Reply:
I changed the default value of `extraConfig` for `ccacheStdenv` from an empty string to null to use as a marker to indicate that the field hasn't been overwritten, so there shouldn't be any problem with this check.

Although, this means that if null is `extraConfig` is explicitly overridden with null, the default `extraConfig` will still be used from `ccacheWrapper` which may be confusing. I will try to make this work without a null marker.